src/table/src/EditableCellField.js

```diff
@@ -1,44 +1,81 @@
-import React, { memo, useRef, useState, useEffect } from 'react'
+import React, { memo, useRef, useState, useMemo, useCallback, useLayoutEffect } from 'react'
 import PropTypes from 'prop-types'
+import { useLatest } from '../../hooks'
 import { Textarea } from '../../textarea'
 
-const EditableCellField = memo(function EditableCellField(props) {
-  const { getTargetRef } = props
-
-  const getTableBodyRef = () => {
-    let ref = getTargetRef()
+function getTableBodyRef(currentRef) {
+  let ref = currentRef
 
-    if (!ref) return
+  if (!ref) return
 
-    while (ref) {
-      const isTableBody = ref.hasAttribute('data-evergreen-table-body')
-      if (isTableBody) {
-        return ref
-      }
-
-      if (ref.parentElement) {
-        ref = ref.parentElement
-      } else {
-        return null
-      }
+  while (ref) {
+    const isTableBody = ref.hasAttribute('data-evergreen-table-body')
+    if (isTableBody) {
+      return ref
     }
 
-    return ref
+    if (ref.parentElement) {
+      ref = ref.parentElement
+    } else {
+      return null
+    }
   }
 
+  return ref
+}
+
+const EditableCellField = memo(function EditableCellField(props) {
+  const { minHeight = 40, minWidth = 80, size, value, zIndex } = props
+
   const latestAnimationFrame = useRef()
   const textareaRef = useRef()
   const tableBodyRef = useRef()
-  const [{ height, left, top, width }, setDimensions] = useState({
-    top: 0,
-    left: 0,
-    height: 0,
-    width: 0
-  })
+  const onCancelRef = useLatest(props.onCancel)
+  const onChangeCompleteRef = useLatest(props.onChangeComplete)
+  const getTargetRef = useLatest(props.getTargetRef)
+  const [height, setHeight] = useState(0)
+  const [width, setWidth] = useState(0)
+  const [left, setLeft] = useState(0)
+  const [top, setTop] = useState(0)
+
+  const update = useCallback(() => {
+    function updater() {
+      const targetRef = getTargetRef.current()
+      if (!targetRef) return
+      tableBodyRef.current = getTableBodyRef(targetRef)
+
+      const {
+        height: targetHeight,
+        left: targetLeft,
+        top: targetTop,
+        width: targetWidth
+      } = targetRef.getBoundingClientRect()
+
+      let calculatedTop
+      if (tableBodyRef.current) {
+        const bounds = tableBodyRef.current.getBoundingClientRect()
+        calculatedTop = Math.min(Math.max(targetTop, bounds.top), bounds.bottom - targetHeight)
+      } else {
+        calculatedTop = targetTop
+      }
+
+      setHeight(targetHeight)
+      setWidth(targetWidth)
+      setLeft(targetLeft)
+      setTop(calculatedTop)
+
+      // recursively run the updater
+      latestAnimationFrame.current = requestAnimationFrame(() => updater())
+    }
+
+    // kick off the updater
+    updater()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
This could be replaced with a hook – given a ref, we recursively find the dimensions using resize observer. Using resize observer would mean no IE support, but I highly doubt we have great IE support already (and we don't support it in the app).

Reply:
plus the perf on it is likely much better. im in favor of resize observer